src/renderers/shared/reconciler/ReactMultiChild.js

```diff
@@ -209,20 +211,23 @@ var ReactMultiChild = {
     },
 
     _reconcilerUpdateChildren: function(prevChildren, nextNestedChildrenElements, transaction, context) {
+      var nextChildren;
       if (__DEV__) {
         if (this._currentElement) {
           try {
             ReactCurrentOwner.current = this._currentElement._owner;
-            return ReactChildReconciler.updateChildren(
-              prevChildren, nextNestedChildrenElements, transaction, context
-            );
+            nextChildren = flattenChildren(nextNestedChildrenElements);
           } finally {
             ReactCurrentOwner.current = null;
           }
         }
+        return ReactChildReconciler.updateChildren(
+          prevChildren, nextChildren, transaction, context
+        );
       }
+      nextChildren = flattenChildren(nextNestedChildrenElements);
```

Comment:
I guess you just moved the callsite, but this actually increases the surface area where flattened children are expected, and ideally we'd decrease it so ReactChildReconciler never needs to flatten.